public/app/features/explore/SupplementaryResultError.tsx

```diff
@@ -1,24 +1,28 @@
 import React, { useState } from 'react';
 
 import { DataQueryError } from '@grafana/data';
-import { Alert, Button } from '@grafana/ui';
+import { Alert, AlertVariant, Button } from '@grafana/ui';
 
 type Props = {
-  error: DataQueryError;
+  error?: DataQueryError;
   title: string;
+  severity?: AlertVariant;
+  suggestedAction?: string;
+  onSuggestedAction?(): void;
+  onRemove?(): void;
 };
 export function SupplementaryResultError(props: Props) {
   const [isOpen, setIsOpen] = useState(false);
   const SHORT_ERROR_MESSAGE_LIMIT = 100;
-  const { error, title } = props;
+  const { error, title, suggestedAction, onSuggestedAction, onRemove, severity = 'warning' } = props;
   // generic get-error-message-logic, taken from
   // /public/app/features/explore/ErrorContainer.tsx
-  const message = error.message || error.data?.message || '';
+  const message = error?.message || error?.data?.message || '';
   const showButton = !isOpen && message.length > SHORT_ERROR_MESSAGE_LIMIT;
 
   return (
-    <Alert title={title} severity="warning">
-      {showButton ? (
+    <Alert title={title} severity={severity} onRemove={onRemove}>
+      {showButton && message ? (
```

Comment:
Just a nit, I think this additional check here is not needed, because it's already in `showButton`:

```
const showButton = !isOpen && message.length > SHORT_ERROR_MESSAGE_LIMIT;
```